src/modules/hip/kernel/box_filter.hpp

```diff
@@ -0,0 +1,775 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+
+__device__ void box_filter_3x3_row_hip_compute(uchar *srcPtr, d_float8 *dst_f8)
+{
+    uint src;
+    float src_f;
+    uint *src_uchar4 = (uint *)srcPtr;
```

Comment:
please use uint2 to read 8 bytes at a time, and modify the below code accordingly.

Reply:
You can pass source as uint * and use directly in the code

Reply:
Done. Using uint3 or uint4 depending on filter size and number of pixels needed for the filter.